src/net_processing.cpp

```diff
@@ -3843,23 +3853,28 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             return;
         }
 
+        const CBlockIndex *pindex;
+        bool is_above_blocktxn_depth;
         {
             LOCK(cs_main);
 
-            const CBlockIndex* pindex = m_chainman.m_blockman.LookupBlockIndex(req.blockhash);
+            pindex = m_chainman.m_blockman.LookupBlockIndex(req.blockhash);
             if (!pindex || !(pindex->nStatus & BLOCK_HAVE_DATA)) {
                 LogPrint(BCLog::NET, "Peer %d sent us a getblocktxn for a block we don't have\n", pfrom.GetId());
                 return;
             }
 
-            if (pindex->nHeight >= m_chainman.ActiveChain().Height() - MAX_BLOCKTXN_DEPTH) {
-                CBlock block;
-                bool ret = ReadBlockFromDisk(block, pindex, m_chainparams.GetConsensus());
-                assert(ret);
-
-                SendBlockTransactions(pfrom, *peer, block, req);
+            is_above_blocktxn_depth = pindex->nHeight >= m_chainman.ActiveChain().Height() - MAX_BLOCKTXN_DEPTH;
+        }
+        if (is_above_blocktxn_depth) {
+            CBlock block;
+            if (!ReadBlockFromDisk(block, pindex, m_chainparams.GetConsensus())) {
+                LogPrint(BCLog::NET, "Cannot load block from disk. It was likely pruned before we could read it.\n");
```

Comment:
I don't think this can happen. MAX_BLOCKTXN_DEPTH is 10, so could keep the assert or at least make it a error log category +  Assume(false)?

Reply:
Could it not happen in regtest?

Reply:
Not sure, but it seems fine to crash in tests if they violate assumptions

Reply:
See `MIN_BLOCKS_TO_KEEP`, which is also enforced in regtest

Reply:
Updated this to keep the assert. Also changed it to take `FlatFilePos` in `ReadBlockFromDisk` so it doesn't have to reacquire the lock again after releasing it.